### PR TITLE
RN0.29 compatability

### PR DESCRIPTION
**Pull request by locly**
React native 0.29 has some changes to the way they initialize libraries, which means they are initialized from the `Application` rather than the `MainActivity`. We added the Activity as a constructor argument to `MapsPackage` in [PR#276](https://github.com/lelandrichardson/react-native-maps/pull/276) to fix Sony Xperia devices with Android 6.0 crashing #271.

After some experimentation and testing we've found that if you supply `ReactContext.getBaseContext()` to the Google Maps API you can get away with not needing the MainActivity. I have also left the `public MapsPackage(Activity activity) { }` constructor for backwards compatibility, but if you'd rather this was removed, let me know and I'll update the PR.

I've also updated the `build.gradle` file as per #359 to get the lib building with RN0.29.

Tested with:
Nexus 6P
Nexus 5
Sony Xperia z5 compact

### CODE REVIEW

**Comment by shtefanntz** [comment]
Thanks for the PR! 
it would be great if this branch would be published to NPM so we can use it with rn0.29

**Comment by julien-rodrigues** [comment]
Hey, 

should the path of the file to update be pointing to `MainApplication.java` ?

https://github.com/lelandrichardson/react-native-maps/pull/363/files#diff-11d8910b58f5c70ccb3bedc0f37c892fR60

**Comment by christopherdro** [comment]
Oops! I didn't see this and ended up submitting #366

**Comment by christopherdro** [comment]
I don't think you need to pass context through again and can use ThemeReactContext. Take a look at my PR if you get a chance. Maybe we can consolidate both our ideas into this PR and close mine out?

**Comment by christopherdro** [comment]
@julien-rodrigues Yes, the docs would have to be updated for 0.29.

We can use https://github.com/Microsoft/react-native-code-push#plugin-configuration-android 
as an example.

**Comment by julien-rodrigues** [comment]
@christopherdro Indeed it seems a great solution

File: android/lib/build.gradle [hunk 1]
```diff
@@ -1,5 +1,4 @@
 apply plugin: 'com.android.library'
-apply from: 'gradle-maven-push.gradle'
```
Comment (felipecsl) [inline android/lib/build.gradle]: this change seems unrelated

File: android/lib/build.gradle [hunk 2]
```diff
@@ -16,7 +15,7 @@ android {
 }
 
 dependencies {
-  compile 'com.facebook.react:react-native:0.27.2'
+  compile 'com.facebook.react:react-native:+'
```
Comment (felipecsl) [inline android/lib/build.gradle]: use `0.29.0` instead.

File: android/lib/src/main/java/com/airbnb/android/react/maps/AirMapManager.java [hunk 3]
```diff
@@ -70,12 +70,12 @@ protected AirMapView createViewInstance(ThemedReactContext context) {
         reactContext = context;
 
         try {
-            MapsInitializer.initialize(reactActivity);
+            MapsInitializer.initialize(this.appContext);
```
Comment (felipecsl) [inline android/lib/src/main/java/com/airbnb/android/react/maps/AirMapManager.java]: omit `this.`

File: android/lib/src/main/java/com/airbnb/android/react/maps/AirMapManager.java [hunk 4]
```diff
@@ -70,12 +70,12 @@ protected AirMapView createViewInstance(ThemedReactContext context) {
...
         } catch (Exception e) {
             e.printStackTrace();
             emitMapError("Map initialize error", "map_init_error");
         }
-        AirMapView view = new AirMapView(context, reactActivity, this);
+        AirMapView view = new AirMapView(context, this.appContext, this);
```
Comment (felipecsl) [inline android/lib/src/main/java/com/airbnb/android/react/maps/AirMapManager.java]: same

File: android/lib/build.gradle (hunk 2 above)
Comment (felipecsl) [inline android/lib/build.gradle]: For more context:
screen shot 2016-07-08 at 3 43 01 pm

**Comment by lelandrichardson** [comment]
I'd like to get a release out for 0.29, so I'm going to go ahead and merge this. There's some gradle stuff that we might tweak before publishing though.

Thanks @locly for your contribution!

**Comment by npomfret** [comment]
I'm getting an Android compile error now using rn 0.29.1:

```
Error:A problem occurred configuring project ':react-native-maps'.
> Could not resolve all dependencies for configuration ':react-native-maps:_debugCompile'.
   > Could not find com.facebook.react:react-native:0.29.0.
     Searched in the following locations:
         file:/Applications/Android Studio.app/Contents/gradle/m2repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         file:/Applications/Android Studio.app/Contents/gradle/m2repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
         file:/Users/nickpomfret/.m2/repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         file:/Users/nickpomfret/.m2/repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
         https://maven.fabric.io/public/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         https://maven.fabric.io/public/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
         https://jcenter.bintray.com/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         https://jcenter.bintray.com/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
         file:/Users/nickpomfret/Documents/github/datr/app/pomochat/node_modules/react-native/android/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         file:/Users/nickpomfret/Documents/github/datr/app/pomochat/node_modules/react-native/android/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
         http://files.couchbase.com/maven2/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         http://files.couchbase.com/maven2/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
         file:/Users/nickpomfret/android/extras/android/m2repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         file:/Users/nickpomfret/android/extras/android/m2repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
         file:/Users/nickpomfret/android/extras/google/m2repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.pom
         file:/Users/nickpomfret/android/extras/google/m2repository/com/facebook/react/react-native/0.29.0/react-native-0.29.0.jar
```

Any idea what might cause this?

**Comment by julien-rodrigues** [comment]
It's because we now require explicitly version 0.29. As states above
